test/distributed/elastic/rendezvous/c10d_rendezvous_backend_test.py

```diff
@@ -50,43 +78,86 @@ def setUp(self) -> None:
             read_timeout="10",
         )
 
+        _, tmp_path = tempfile.mkstemp()
+
+        # Parameters for filestore testing.
+        self._params_filestore = RendezvousParameters(
+            backend="dummy_backend",
+            endpoint=tmp_path,
+            run_id="dummy_run_id",
+            min_nodes=1,
+            max_nodes=1,
+            store_type="fIlE",
+        )
+        self._expected_endpoint_file = tmp_path
+        self._expected_temp_dir = tempfile.gettempdir()
+
         self._expected_endpoint_host = "localhost"
         self._expected_endpoint_port = 29300
         self._expected_store_type = TCPStore
         self._expected_read_timeout = timedelta(seconds=10)
 
-    def test_create_backend_returns_backend(self) -> None:
+    def tearDown(self) -> None:
+        os.remove(self._expected_endpoint_file)
+
+
+    def _run_test_with_store(self, store_type: str, test_to_run: Callable):
+        """
+        Use this function to specify the store type to use in a test. If
+        not used, the test will default to TCPStore.
+        """
+        if store_type == "file":
+            self._params = self._params_filestore
+            self._expected_store_type = FileStore
+            self._expected_read_timeout = timedelta(seconds=300)
+            # in case there is a test which defaults to tcp store, we set the
+            # default tcp parameters
+            self._expected_endpoint_port = 29400
```

Comment:
Nit: Is there such a test case? I would assume that if the store type is `file` and somehow we use a TCP port, it is likely a bug.